services/port_manager/src/main/java/com/futurewei/alcor/portmanager/proxy/IpManagerProxy.java

```diff
@@ -84,27 +84,47 @@ private void addIpAddrRollback(AbstractIpAddrRollback rollback, IpAddrRequest ip
         rollbacks.push(rollback);
     }
 
-    public List<IpAddrRequest> allocateIpAddress(Object args) throws Exception {
+    /**
+     * Allocate a random ipv4 and ipv6 address from ip manager service
+     * @param args PortState
+     * @return A list of IpAddrRequest
+     * @throws Exception Rest request exception
```

Comment:
@chenpiaoping, my previous comment may not be 100% clear. I meant that IPv4 only and Ipv4v6 should both be supported, with IPv4 only the default option.

Reply:
I see, should i revert it?

Reply:
Those are good change. Maybe by default we only trigger Ipv4 allocation, but add one more flag to trigger additional Ipv6 allocation. 

Neutron may or may not have this field. Let me quickly check.

Reply:
I think if there is ipv6 subnet in the vpc, the ipv6 address will be assigned.

Reply:
My current implementation may not be very reasonable.

Reply:
It is totally fine, we do the development and design in an iterative/agile way: When we find it something not right, we change it quickly. 

My current thinking is: This should be a consistent behavior on VPC level. If a VPC allows Ipv6, every of its Subnets will have Ipv6 ranges therefore trigger Ipv4v6 allocation here.

In our Subnet Object, there is a Ipv4RangeId and a Ipv6RangeId. Therefore our logic could be that if the Ipv6RangeId is not null, we allocate Ipv6 here. Does this make sense?

Reply:
Agree with you, but i think i need to make some changes in ip manager.

Reply:
It's time for me to get off work, will do it tomorrow.

Reply:
I see. Might not be a small change considering here requires a full set of UTs. 

Let us comment out Ipv6 allocation part in this PR, and continue to improve it in next PR.